scripts/build/build_darwin_framework.py

```diff
@@ -42,23 +42,20 @@ def build_darwin_framework(args):
     if not os.path.exists(abs_path):
         os.mkdir(abs_path)
 
-    sdk = 'macosx' if os.environ.get('SDKROOT') == None else os.environ.get('SDKROOT')
-    arch = platform.machine() if sdk.startswith('macosx') else 'arm64'
-
     command = [
         'xcodebuild',
         '-scheme',
         args.target,
         '-sdk',
-        sdk,
+        args.target_sdk,
```

Comment:
Could we convert the logic from this file into build.py (that uses targets.py/builders) or move it out of scripts/build please?

This is a completely separate and stand-alone build system. It seems added in #20337 and with fast track without me having reviewed it (I was not aware it exists). 

Looking at it, this is a stand-alone binary, it is not executable (should be - easy fix) and it is not using the rest of the `build` infra (that includes unit tests for content and output).

Reply:
Created #23229

I will give a checkmark on this since this comment has nothing to do with this PR itself, however @krypton36  (or @bzbarsky-apple / @vivien-apple ) please make the updates as a short term followup.

Reply:
I agree it would be nice to have this file to be part of build.py and more generally use the gn build system as much as possible. I need to spend more time looking how things are glued together inside `build.py` though.